src/commonTest/kotlin/app/cash/turbine/FlowTest.kt

```diff
@@ -727,4 +733,25 @@ class FlowTest {
       awaitComplete()
     }
   }
+
+  @Test
+  fun timeoutsAreCaptured() = runTest {
+    flow<Nothing> {
+      withTimeout(500) {
+        delay(2000)
+      }
+    }.test {
+      assertTrue(awaitError() is TimeoutCancellationException)
+    }
+  }
+
+  @Test
+  fun cancellationsAreCaptured() = runTest {
```

Comment:
Not sure how I feel about this behavior, but... gut feeling is that this is less wrong to me than the alternatives.